src/choices.c

```diff
@@ -23,6 +23,9 @@ min_match_length(char *str, char *query)
 
 	for (mlen = 0, mstart = 0; str[mstart] != '\0'; ++mstart)
 		if (tolower(str[mstart]) == tolower(query[0])) {
+			if (*start_pos == -1)
+				*start_pos = mstart;
```

Comment:
This is just a flag to test if `start_pos` has already been set. The issue here is that it could be reset if the word being searched contained the first character of the query more than once.